File: tareas.js
class Tarea {
  constructor(id, descripcion, prioridad = 'normal') {
    const prioridadesValidas = ['urgente', 'normal'];
    if (!prioridadesValidas.includes(prioridad)) {
      throw new Error("Prioridad inválida. Debe ser 'urgente' o 'normal'.");
    }

    this.id = id;
    this.descripcion = descripcion;
    this.prioridad = prioridad;
    this.completada = false;
  }

  completar() {
    this.completada = true;
  }
}


class GestorTareas {
  constructor() {
    this.tareas = [];
  }

  crearTarea(descripcion, prioridad = 'normal') {
    const id = Date.now();
    const nueva = new Tarea(id, descripcion, prioridad);
    this.tareas.push(nueva);
    this.ordenarPorPrioridad();
    return nueva;
  }

  eliminarTarea(id) {
    this.tareas = this.tareas.filter(t => t.id !== id);
  }

  completarTarea(id) {
    const tarea = this.buscarPorId(id);
    if (tarea) tarea.completar();
  }

  buscarPorId(id) {
    return this.tareas.find(t => t.id === id);
  }

  filtrarPorPrioridad(prioridad) {
    return this.tareas.filter(t => t.prioridad === prioridad);
  }

  ordenarPorPrioridad() {
    this.tareas.sort((a, b) => {
      if (a.prioridad === 'urgente' && b.prioridad !== 'urgente') return -1;
      if (a.prioridad !== 'urgente' && b.prioridad === 'urgente') return 1;
      return 0;
    });
  }

  mostrarTareas() {
    console.log("=== Lista de Tareas ===");
    this.tareas.forEach(t => {
      console.log(`[${t.completada ? "✔" : " "}] (${t.prioridad.toUpperCase()}) ${t.descripcion} [ID: ${t.id}]`);
    });
  }
}

File: tareas.py
class Tarea:
    def __init__(self, id_tarea, descripcion, prioridad):
        self.id_tarea = id_tarea
        self.descripcion = descripcion
        self.prioridad = 0 if prioridad == "urgente" else 1  # 0 es urgente, 1 es normal
        self.estado = "pendiente"

    def completar(self):
        """Marca la tarea como completada."""
        self.estado = "completada"

    def __lt__(self, otra_tarea):
        """Permite comparar tareas por prioridad en la cola."""
        return self.prioridad < otra_tarea.prioridad  # Menor valor es mayor prioridad

    def __repr__(self):
        return f"Tarea(ID: {self.id_tarea}, Descripción: {self.descripcion}, Prioridad: {'Urgente' if self.prioridad == 0 else 'Normal'}, Estado: {self.estado})"
